cirq/contrib/acquaintance/gates.py

```diff
@@ -0,0 +1,311 @@
+        layers['intra'].append(shift)
+    else:
+        # before
+        if left_size == acquaintance_size - 1:
+            # right part
+            pre_acquaintance_gate = ACQUAINT(*qubits[:acquaintance_size])
+            acquaint_insides(
+                    swap_gate=swap_gate,
+                    acquaintance_gate=pre_acquaintance_gate,
+                    qubits=right_part,
+                    before=True,
+                    layers=layers,
+                    mapping=mapping)
+
+        if right_size == acquaintance_size - 1:
+            # left part
+            pre_acquaintance_gate = ACQUAINT(*qubits[-acquaintance_size:])
+            acquaint_insides(
+                    swap_gate=swap_gate,
+                    acquaintance_gate=pre_acquaintance_gate,
+                    qubits=left_part[::-1],
+                    before=True,
+                    layers=layers,
+                    mapping=mapping)
+
+        layers['intra'].append(shift)
+        shift.gate.update_mapping(mapping, qubits)
+
+        # after
+        if ((left_size == acquaintance_size - 1) and
+            (right_size > 1)):
+            # right part
+            post_acquaintance_gate = ACQUAINT(*qubits[-acquaintance_size:])
+
+            new_left_part = qubits[right_size - 1::-1]
+            acquaint_insides(
+                    swap_gate=swap_gate,
+                    acquaintance_gate=post_acquaintance_gate,
+                    qubits=new_left_part,
+                    before=False,
+                    layers=layers,
+                    mapping=mapping)
+
+        if ((right_size == acquaintance_size - 1) and
+            (left_size > 1)):
+            # left part
+
+            post_acquaintance_gate = ACQUAINT(*qubits[:acquaintance_size])
+            acquaint_insides(
+                    swap_gate=swap_gate,
+                    acquaintance_gate=post_acquaintance_gate,
+                    qubits=qubits[right_size:],
+                    before=False,
+                    layers=layers,
+                    mapping=mapping)
+
+
+class SwapNetworkGate(CompositeGate, PermutationGate):
+    """A single gate representing a generalized swap network.
+
+    Args:
+        part_lens: An sequence indicating the sizes of the parts in the
+            partition defining the swap network.
+        acquaintance_size: An int indicating the locality of the logical gates
+            desired; used to keep track of this while nesting. If 0, no
+            acquaintance gates are inserted.
+
+    Attributes:
+        part_lens: See above
+        acquaintance_size: See above.
+        swap_gate: The gate used to swap logical indices.
+    """
+
+    def __init__(self,
+                 part_lens: Sequence[int],
+                 acquaintance_size: int=0,
+                 swap_gate: Gate=SWAP
+                 ) -> None:
+        if len(part_lens) < 2:
+            raise ValueError('part_lens must have length at least 2.')
```

Comment:
part_lens isn't an int, it can't be at least 2.

Here's a trick for writing most of these kinds of error messages: just repeat the condition.  `raise ValueError('len(part_lens) < 2')`

Reply:
`part_lens` is a list of ints, so it's length can be at least 2.

Reply:
My bad, I misread the message as seeing "must be at least".